### PR TITLE
feat: bulk delete configs support

### PR DESCRIPTION
**Pull request by SrikarMannepalli**
## Description
This PR adds the API contracts to support bulk delete of configs 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto [hunk 1]
```diff
@@ -119,6 +121,31 @@ message DeleteConfigResponse {
...
+    // optional - only required if config applies to a specific context.
+    // If empty, specified config is associated with a default context.
+    string context = 3;
+  }
+}
+
+message DeleteConfigsResponse {
+  repeated DeletedConfig deleted_configs = 1;
+
+  message DeletedConfig {
```
Comment (aaron-steinfeld) [inline config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto:141]: The singular API returns a `ContextSpecificConfig` - why not reuse that?

Comment (SrikarMannepalli) [reply config-service-api/src/main/proto/org/hypertrace/config/service/v1/config_service.proto]: I was contemplating between using it itself and a separate config, but went the way of `UpsertAllConfigs` finally. Updated it now.

**Review by skjindal93 — DISMISSED** [review]

**Review by skjindal93 — DISMISSED** [review]

File: config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java [hunk 2]
```diff
@@ -140,6 +144,49 @@ public void deleteConfig(
...
+        return;
+      }
+      Map<ConfigResourceContext, Value> valuesByContext =
+          request.getConfigsList().stream()
+              .map(
+                  requestedDelete ->
+                      Map.entry(this.getConfigResourceContext(requestedDelete), emptyValue()))
+              .collect(ImmutableMap.toImmutableMap(Entry::getKey, Entry::getValue));
+
+      List<UpsertedConfig> deletedConfigs =
```
Comment (aaron-steinfeld) [inline config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java:165]: Wouldn't this be returning the post-delete values (aka empty). The response from this API I would expect would be the values that have been deleted, so if we're not going to return that we should be modifying the API.

Comment (aaron-steinfeld) [reply config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java:165]: I take it back - I see it returns the prev config too :)

File: config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java [hunk 3]
```diff
@@ -140,6 +144,49 @@ public void deleteConfig(
...
+      responseObserver.onNext(
+          DeleteConfigsResponse.newBuilder()
+              .addAllDeletedConfigs(
+                  deletedConfigs.stream()
+                      .map(this::buildDeletedContextSpecificConfig)
+                      .collect(Collectors.toUnmodifiableList()))
+              .build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      log.error("Delete configs failed for request: {}", request, e);
```
Comment (aaron-steinfeld) [inline config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java:177]: responseObserver.onError missing here.

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**